src/main/java/elberger/maze/Maze.java

```diff
@@ -114,6 +90,35 @@ else if (currentCell.getX() == nextCell.getX() - 1)
 		}
 	}
 
+	public void searchPath(Cell cell)
+	{
+		Stack<Cell> stack = new Stack<>();
+		stack.push(cell);
+		cell.setVisited(true);
+
+		while (!stack.isEmpty())
+		{
+			Cell currentCell = stack.pop();
+			Cell nextCell = getNeighbors(currentCell).get(0);
+
+			if (!nextCell.isVisited())
+			{
+				nextCell.setVisited(true);
+				breakWalls(currentCell, nextCell);
+				stack.push(nextCell);
+			}
+			else if (nextCell.isVisited())
```

Comment:
by definition if the first if statement was false, then this if statement would be true